src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
+		{
+			Grab.Remove(window);
+			Gdk.Pointer.Ungrab(CURRENT_TIME);
+			Gdk.Keyboard.Ungrab(CURRENT_TIME);
+		}
+	}
+
+	public partial class DatePickerWindow : Window
+	{
+		VBox _datebox;
+		RangeCalendar _calendar;
+		public delegate void DateEventHandler(object sender, DateEventArgs args);
+
+		public event DateEventHandler? OnDateTimeChanged;
+
+		public DatePickerWindow()
+			: base(WindowType.Popup)
+		{
+			Title = "DatePicker";
+			TypeHint = Gdk.WindowTypeHint.Desktop;
+			WindowPosition = WindowPosition.Mouse;
+			BorderWidth = 1;
+			Resizable = false;
+			Decorated = false;
+			DestroyWithParent = true;
+			SkipPagerHint = true;
+			SkipTaskbarHint = true;
+
+			_datebox = new VBox();
+			_datebox.Spacing = 6;
+			_datebox.BorderWidth = 3;
+
+			_calendar = new RangeCalendar();
+			_calendar.CanFocus = true;
+			_calendar.DisplayOptions = CalendarDisplayOptions.ShowHeading;
+			_datebox.Add(_calendar);
+			Box.BoxChild dateBoxChild = ((Box.BoxChild)(_datebox[_calendar]));
+			dateBoxChild.Position = 0;
+
+			Add(_datebox);
+
+			if ((Child != null))
+			{
+				Child.ShowAll();
+			}
+
+			Show();
+			ButtonPressEvent += new ButtonPressEventHandler(OnButtonPressEvent);
+			_calendar.ButtonPressEvent += new ButtonPressEventHandler(OnCalendarButtonPressEvent);
+			_calendar.DaySelected += new EventHandler(OnCalendarDaySelected);
+			_calendar.DaySelectedDoubleClick += new EventHandler(OnCalendarDaySelectedDoubleClick);
+			GrabHelper.GrabWindow(this);
+			SelectedDate = DateTime.Now;
+		}
+
+		public DateTime SelectedDate
+		{
+			get
+			{
+				return _calendar.Date;
+			}
+
+			set
+			{
+				_calendar.Date = value;
+			}
+		}
+
+		public DateTime MinimumDate
+		{
+			get
+			{
+				return _calendar.MinimumDate;
+			}
+
+			set
+			{
+				_calendar.MinimumDate = value;
+			}
+		}
+
+		public DateTime MaximumDate
+		{
+			get
+			{
+				return _calendar.MaximumDate;
+			}
+
+			set
+			{
+				_calendar.MaximumDate = value;
+			}
+		}
+
+		protected virtual void OnButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			CloseCalendar();
+		}
+
+		protected virtual void OnCalendarButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			args.RetVal = true;
+		}
+
+		protected virtual void OnCalendarDaySelected(object? sender, EventArgs e)
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+		}
+
+		protected virtual void OnCalendarDaySelectedDoubleClick(object? sender, EventArgs e)
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+			CloseCalendar();
+		}
+
+		void CloseCalendar()
+		{
+			GrabHelper.RemoveGrab(this);
+			Destroy();
+		}
+
+		void NotifyDateChanged()
+		{
+			OnDateTimeChanged?.Invoke(this, new DateEventArgs(SelectedDate));
+		}
+
+		class RangeCalendar : Calendar
+		{
+			DateTime _minimumDate;
+			DateTime _maximumDate;
+
+			public RangeCalendar()
+			{
+				_minimumDate = new DateTime(1900, 1, 1);
+				_maximumDate = new DateTime(2199, 1, 1);
```

Comment:
@parhamsaremi is XamarinForms' DatePicker hardcoding these values above too?

Reply:
Yes: https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Controls/DatePicker.cs#L171